lib/vector-core/src/event/log_event.rs

```diff
@@ -150,7 +151,8 @@ impl LogEvent {
     /// valid for `LogNamespace::Legacy`
     pub fn from_str_legacy(msg: impl Into<String>) -> Self {
         let mut log = LogEvent::default();
-        log.insert(log_schema().message_key(), msg.into());
+        log.maybe_insert(PathPrefix::Event, log_schema().message_key(), msg.into());
```

Comment:
suggestion: Now that you've went and refactored all these, it might be worth adding another helper to specifically encode this one pattern into something like `log.maybe_insert_message(msg)`